tasks/managed/filter-published-fbc-images/filter-published-fbc-images.yaml

```diff
@@ -0,0 +1,505 @@
+              target_index_to_components[$target_index]="$i"
+              unique_target_indices+=("$target_index")
+            else
+              target_index_to_components[$target_index]+=" $i"
+            fi
+          fi
+        done
+        
+        if [ "$all_components_without_target" = true ]; then
+          echo ""
+          echo "ℹ️  No targetIndex found in any component"
+          echo "  This is expected for:"
+          echo "    - Stage/development builds"
+          echo "    - Builds not targeting a specific index"
+          echo "  → Skipping idempotence filtering (keeping all components)"
+          cp "${SNAPSHOT_FILE}" "${FILTERED_SNAPSHOT_FILE}"
+          echo -n "filtered-snapshot.json" > "$(results.filteredSnapshotPath.path)"
+          exit 0
+        fi
+        
+        unique_count=${#unique_target_indices[@]}
+        echo ""
+        echo "Found $unique_count unique targetIndex value(s)"
+        echo ""
+        
+        case "$pyxis_server" in
+          production)
+            PYXIS_URL="https://pyxis.api.redhat.com/"
+            ;;
+          stage)
+            PYXIS_URL="https://pyxis.preprod.api.redhat.com/"
+            ;;
+          production-internal)
+            PYXIS_URL="https://pyxis.engineering.redhat.com/"
+            ;;
+          stage-internal)
+            PYXIS_URL="https://pyxis.stage.engineering.redhat.com/"
+            ;;
+          *)
+            echo "ERROR: Invalid pyxis server: $pyxis_server"
+            exit 1
+            ;;
+        esac
+
+        echo "Pyxis server: $PYXIS_URL"
+        echo ""
+        
+        # Calculate date filter (90 days ago) for Pyxis query optimization
+        # Format: YYYY-MM-DD (Pyxis expects ISO 8601 date format)
+        date_filter=$(date -u -d '90 days ago' '+%Y-%m-%d' 2>/dev/null || date -u -v-90d '+%Y-%m-%d')
+        echo "Date filter: last_update_date>=$date_filter (last 90 days)"
+        echo ""
+        
+        # Function to query Pyxis and extract published fragments for a targetIndex
+        query_pyxis_for_target_index() {
+          local target_index="$1"
+          local max_retries=3
+          local retry_count=0
+          local index_response=""
+          local http_code=""
+          
+          echo "  Querying Pyxis for targetIndex: ${target_index}"
+          
+          while [ $retry_count -le $max_retries ]; do
+            if [ $retry_count -gt 0 ]; then
+              echo "  Retry attempt $retry_count/$max_retries..."
+              sleep $((retry_count * 2))  # Exponential backoff: 2s, 4s, 6s
+            fi
+            
+            local response_file
+            local http_code_output
+            response_file=$(mktemp)
+            http_code_output=$(mktemp)
+            
+            # Build Pyxis query URL with optimization parameters:
+            # - docker_image_id filter: find indices for this catalog
+            # - last_update_date filter: only check indices updated in last 90 days
```

Comment:
Why do we need 90 days of results? So far I see issues with people trying to re-run release the release in a very short period. I bet  30 days would be already a lot (to be conservative). WDYT @mmalina

Reply:
You're absolutely right! Thanks! 
changed from 90 days to 30 days (my bad, I always use 90 by default for some reason)

Reply:
I changed it to 30 days. Let's try a smaller value first
We can always increase it if needed